File: Spark1.x/Spark-kafka-streaming-1.6_java/src/main/java/com/hwx/StreamTest.java
package com.hwx;


import org.apache.kafka.clients.consumer.ConsumerConfig;
import org.apache.kafka.clients.consumer.ConsumerRecord;
import org.apache.kafka.common.serialization.StringDeserializer;
import org.apache.spark.SparkConf;
import org.apache.spark.SparkEnv;
import org.apache.spark.TaskContext;
import org.apache.spark.api.java.JavaRDD;
import org.apache.spark.api.java.JavaSparkContext;
import org.apache.spark.api.java.function.Function;
import org.apache.spark.api.java.function.VoidFunction;
import org.apache.spark.storage.StorageLevel;
import org.apache.spark.streaming.Durations;
import org.apache.spark.streaming.api.java.JavaDStream;
import org.apache.spark.streaming.api.java.JavaInputDStream;
import org.apache.spark.streaming.api.java.JavaStreamingContext;
import org.apache.spark.streaming.kafka010.ConsumerStrategies;
import org.apache.spark.streaming.kafka010.KafkaUtils;
import org.apache.spark.streaming.kafka010.LocationStrategies;

import java.util.*;


public class StreamTest
{
    public static void main( String[] args )
    {

        SparkConf conf = new SparkConf().setAppName("StreamTest");
        JavaSparkContext sc = new JavaSparkContext(conf);
        JavaStreamingContext jssc = new JavaStreamingContext(sc,  Durations.seconds(Integer.parseInt(args[2])));

        Map<String, Object> kafkaParams = new HashMap<String, Object>();
        kafkaParams.put(ConsumerConfig.BOOTSTRAP_SERVERS_CONFIG, args[0]);
        kafkaParams.put(ConsumerConfig.GROUP_ID_CONFIG, "GroupStreamTest");
        kafkaParams.put(ConsumerConfig.KEY_DESERIALIZER_CLASS_CONFIG, StringDeserializer.class);
        kafkaParams.put(ConsumerConfig.VALUE_DESERIALIZER_CLASS_CONFIG, StringDeserializer.class);
        Set<String> topicsSet = new HashSet<String>(Arrays.asList(args[1]));
        JavaInputDStream<ConsumerRecord<Object, Object>> stream =  KafkaUtils.createDirectStream(
            jssc, LocationStrategies.PreferConsistent(), ConsumerStrategies.Subscribe(topicsSet,kafkaParams));

        JavaDStream<String> lines = stream.map(new Function<ConsumerRecord<Object,Object>, String>() {

            public String call(ConsumerRecord<Object, Object> objectObjectConsumerRecord) throws Exception {
                return objectObjectConsumerRecord.value().toString();
            }
        });

        lines.foreachRDD(new VoidFunction<JavaRDD<String>>() {
            public void call(JavaRDD<String> stringJavaRDD) throws Exception {
                stringJavaRDD.foreachPartition(new VoidFunction<Iterator<String>>() {
                    public void call(Iterator<String> stringIterator) throws Exception {
                        TaskContext tc = TaskContext.get();
                        System.out.println("Executor id : " + SparkEnv.get().executorId() + "  Task ID:  " + tc.taskAttemptId());

                    }
                });

            }
        });

        lines.print();
        // Start the computation
        jssc.start();
        jssc.awaitTermination();



    }
}

File: Spark2/Spark-kafka-streaming/src/main/scala/com/hwx/StreamTest.scala
package com.hwx

import org.apache.spark.SparkConf
import org.apache.spark.streaming.{Seconds, StreamingContext}


object StreamTest extends App {

  val conf = new SparkConf().setAppName("StreamTest")

  import org.apache.kafka.common.serialization.StringDeserializer
  import org.apache.spark.streaming.kafka010._
  import org.apache.spark.streaming.kafka010.LocationStrategies.PreferConsistent
  import org.apache.spark.streaming.kafka010.ConsumerStrategies.Subscribe

  val kafkaParams = Map[String, Object](
    "bootstrap.servers" -> args(0),
    "key.deserializer" -> classOf[StringDeserializer],
    "value.deserializer" -> classOf[StringDeserializer],
    "group.id" -> "StreamTestGroup",
    "auto.offset.reset" -> "latest",
    "enable.auto.commit" -> (false: java.lang.Boolean)
  )
  val topics = Array(args(1))
  val streamingContext = new StreamingContext(conf, Seconds(Integer.parseInt(args(2))))
  val stream = KafkaUtils.createDirectStream[String, String](
    streamingContext,
    PreferConsistent,
    Subscribe[String, String](topics, kafkaParams)
  )

  stream.map(record => (record.key, record.value)).print()


  stream.foreachRDD { rdd =>
    val offsetRanges = rdd.asInstanceOf[HasOffsetRanges].offsetRanges


    stream.asInstanceOf[CanCommitOffsets].commitAsync(offsetRanges)
  }

  // Start the context
  streamingContext.start()
  streamingContext.awaitTermination()

}

